pkg/storage/pebble_batch.go

```diff
@@ -549,6 +549,30 @@ func (p *pebbleBatch) Commit(sync bool) error {
 	return err
 }
 
+// CommitNoSyncWait implements the Batch interface.
+func (p *pebbleBatch) CommitNoSyncWait() error {
+	if p.batch == nil {
+		panic("called with nil batch")
+	}
+	err := p.db.ApplyNoSyncWait(p.batch, pebble.Sync)
+	if err != nil {
+		panic(err)
+	}
+	return err
+}
+
+// SyncWait implements the Batch interface.
+func (p *pebbleBatch) SyncWait() error {
+	if p.batch == nil {
+		panic("called with nil batch")
+	}
+	err := p.batch.SyncWait()
+	if err != nil {
+		panic(err)
```

Comment:
Why return the error then? Same question about `CommitNoSyncWait` and, higher up and in old code, `Commit`. A comment on these would be helpful to understand the rationale.

Reply:
Nice catch! I brought this up with the experts in https://cockroachlabs.slack.com/archives/CAC6K3SLU/p1674678349690899. For now, we're going to keep the confusing code with a trio of TODOs.